primitives/src/sync.rs

```diff
@@ -0,0 +1,149 @@
+use core::ops::{Deref, DerefMut};
```

Comment:
Consider treating this crate as a means to easily swap out standard implementations of `Mutex` rather than copying an existing implementation. 

DRY